src/control/server/config/server.go

```diff
@@ -528,11 +533,14 @@ func (cfg *Server) Validate(log logging.Logger, hugePageSize int, fis *hardware.
 
 		// If the config doesn't specify hugepages, use the minimum. Otherwise, validate
 		// that the configured amount is sufficient.
-		if cfg.NrHugepages == 0 {
+		switch {
+		case cfg.NrHugepages == -1:
+			return FaultConfigHugepagesDisabled
+		case cfg.NrHugepages == 0:
```

Comment:
does this case correspond to both nr_hugepages: 0 specified explicitly in the daos_server.yml file, as well as no nr_hugepages present in the config file?

Reply:
yes, zero-value of field and explicit passing of 0